devices/YarpRobotLoggerDevice/tests/YarpRobotLoggerDeviceTest.cpp

```diff
@@ -0,0 +1,224 @@
+    REQUIRE(yarprobotinterfaceInstance.parsingIsSuccessful);
+
+    // Enter the startup phase, that will open all the devices and  call attach if necessary
+    REQUIRE(yarprobotinterfaceInstance.robot.enterPhase(yarp::robotinterface::ActionPhaseStartup));
+
+    // At this point, the system is running and the thread that called the
+    // enterPhase methods does not need to do anything else
+
+    INFO("yarprobotinterface successfully loaded and started.");
+
+    // Sleep 0.2 second to collect some data
+    std::this_thread::sleep_for(std::chrono::milliseconds(200));
+
+    // Now that some time passed, let's read some values that will compare against the values that we will read from the logger, and then close the robotinterface
+    double sim_joint_1_pos_read = 0.0;
+    double accelerometer_z_read = 0.0;
+
+    // Make sure that the requires devices are there
+    // These strings are the name attributes of the device tags specified in launch-yarp-robot-logger.xml
+    CHECK(yarprobotinterfaceInstance.robot.hasDevice("sim_controlboard"));
+    CHECK(yarprobotinterfaceInstance.robot.hasDevice("sim_imu"));
+
+    // Read encoders
+    yarp::dev::IEncoders* iencs=nullptr;
+    CHECK(yarprobotinterfaceInstance.robot.device("sim_controlboard").driver()->view(iencs));
+    REQUIRE(iencs != nullptr);
+    Eigen::VectorXd jointPosRead;
+    size_t nrOfJoints = 4;
+    jointPosRead.resize(nrOfJoints);
+    CHECK(iencs->getEncoders(jointPosRead.data()));
+
+    // Read accelerometer
+    yarp::dev::IThreeAxisLinearAccelerometers* ilinacc=nullptr;
+    CHECK(yarprobotinterfaceInstance.robot.device("sim_imu").driver()->view(ilinacc));
+    REQUIRE(ilinacc != nullptr);
+    yarp::sig::Vector accelerometerReadYARP;
+    Eigen::VectorXd accelerometerRead;
+    size_t nrOfDirectionsInAccelerometer = 3;
+    accelerometerRead.resize(nrOfDirectionsInAccelerometer);
+    // The device only has one sensor
+    size_t sensIndex = 0;
+    double timestamp;
+    CHECK(ilinacc->getThreeAxisLinearAccelerometerMeasure(sensIndex, accelerometerReadYARP, timestamp));
+    accelerometerRead = yarp::eigen::toEigen(accelerometerReadYARP);
+
+
+    INFO("Halting yarprobotinterface.");
+    REQUIRE(yarprobotinterfaceInstance.robot.enterPhase(yarp::robotinterface::ActionPhaseInterrupt1));
+    REQUIRE(yarprobotinterfaceInstance.robot.enterPhase(yarp::robotinterface::ActionPhaseShutdown));
+
+    INFO("yarprobotinterface successfully halted.");
+
+    // Now that the yarprobotinterface has been halted, let's open our .mat files and check if the log is working
+    currentPath = std::filesystem::current_path();
+    std::string matLogFilename;
+    for (const auto& entry : std::filesystem::directory_iterator(currentPath)) {
+        if (entry.is_regular_file()) {
+            std::filesystem::path filePath = entry.path();
+            std::string extension = filePath.extension().string();
+
+            if (extension == ".mat") {
+                matLogFilename = filePath;
```

Comment:
```suggestion
                matLogFilename = filePath.string();
```

I think this is one cause of issues